src/compiler/moduleNameResolver.ts

```diff
@@ -484,6 +484,13 @@ function getOriginalAndResolvedFileName(fileName: string, host: ModuleResolution
     };
 }
 
+function getCandidateFromTypeRoot(typeRoot: string, typeReferenceDirectiveName: string, moduleResolutionState: ModuleResolutionState) {
+    return combinePaths(typeRoot, endsWith(typeRoot, "/node_modules/@types") || endsWith(typeRoot, "/node_modules/@types/") ?
+        mangleScopedPackageNameWithTrace(typeReferenceDirectiveName, moduleResolutionState) :
+        typeReferenceDirectiveName
+    );

```

Comment:
Can you separate the conditional into its own variable?

Reply:
I'd appreciate this too; I have a hard time reading a conditional as big as this within a call expression.